### ISSUE TITLE
Perpetual protocol - Getting error for unsupported  'NoneType' and 'decimal.Decimal'

### ISSUE THREAD

**Issue by rapcmia** [hummingbot/hummingbot#5693]
**Describe the bug**
When running Perpetual connector for long term, we noticed that after a sudden network change on our AWS instance, the client would return an warning of `WARNING - Error on POST` resulting to logs kept reporting `Unhandled error in background task: unsupported operand type(s) for -: 'NoneType' and 'decimal.Decimal'`

```
2022-09-08 07:45:35,051 - 99724 - GatewayEVMPerpetual - WARNING - Error on POST https://localhost:25393/amm/perp/market-prices Error: {'message': 'Unknown error.', 'httpErrorCode': 500, 'errorCode': 1099, 'stack': 'Error: Unknown error.\n    at /usr/src/app/dist/src/connectors/perp/perp.controllers.js:24:19\n    at Generator.throw (<anonymous>)\n    at rejected (/usr/src/app/dist/src/connectors/perp/perp.controllers.js:6:65)'}

2022-09-08 07:45:35,186 - 99724 - GatewayEVMPerpetual - WARNING - Error on POST https://localhost:25393/amm/perp/market-prices Error: {'message': 'Unknown error.', 'httpErrorCode': 500, 'errorCode': 1099, 'stack': 'Error: Unknown error.\n    at /usr/src/app/dist/src/connectors/perp/perp.controllers.js:24:19\n    at Generator.throw (<anonymous>)\n    at rejected (/usr/src/app/dist/src/connectors/perp/perp.controllers.js:6:65)'}

2022-09-08 07:45:35,187 - 99724 - hummingbot.core.utils.async_utils - ERROR - Unhandled error in background task: unsupported operand type(s) for -: 'NoneType' and 'decimal.Decimal'
Traceback (most recent call last):
  File "/home/ubuntu/perp/hummingbot/core/utils/async_utils.py", line 9, in safe_wrapper
    return await c
  File "/home/ubuntu/perp/hummingbot/strategy/spot_perpetual_arbitrage/spot_perpetual_arbitrage.py", line 207, in main
    proposals = [p for p in proposals if p.profit_pct() >= self._min_opening_arbitrage_pct]
  File "/home/ubuntu/perp/hummingbot/strategy/spot_perpetual_arbitrage/spot_perpetual_arbitrage.py", line 207, in <listcomp>
    proposals = [p for p in proposals if p.profit_pct() >= self._min_opening_arbitrage_pct]
  File "/home/ubuntu/perp/hummingbot/strategy/spot_perpetual_arbitrage/arb_proposal.py", line 59, in profit_pct
    return (sell_price - buy_price) / buy_price
TypeError: unsupported operand type(s) for -: 'NoneType' and 'decimal.Decimal'
```
There are also instances of `ValueError' object is not subscriptable`
```
TypeError: unsupported operand type(s) for -: 'NoneType' and 'decimal.Decimal'
2022-09-08 07:46:03,012 - 99724 - GatewayEVMPerpetual - ERROR - 'ValueError' object is not subscriptable
Traceback (most recent call last):
  File "/home/ubuntu/perp/hummingbot/connector/gateway/amm/gateway_evm_perpetual.py", line 457, in _status_polling_loop
    await safe_gather(
  File "/home/ubuntu/perp/hummingbot/core/utils/async_utils.py", line 22, in safe_gather
    return await asyncio.gather(*args, **kwargs)
  File "/home/ubuntu/perp/hummingbot/connector/gateway/amm/gateway_evm_perpetual.py", line 556, in update_positions
    trading_pair = f"{position['base']}-{position['quote']}"
TypeError: 'ValueError' object is not subscriptable
2022-09-08 07:46:07,011 - 99724 - GatewayEVMPerpetual - WARNING - 'markPrice'
2022-09-08 07:46:07,091 - 99724 - GatewayEVMPerpetual - WARNING - Error on POST 
```
Tried to replicate on local connection by disconnecting and got the errors instantly. Tried to leave the client for 3-4hours. Noticed that it was able to found an arbitrage but this time we are able to reproduce the issue related to `ValueError' object is not subscriptable` 

![image](https://user-images.githubusercontent.com/73840223/189560857-03b11e9b-ee7a-408b-b4ee-be832871678b.png)

**Impact**
We observed that the strategy and perp protocol is able to work since its trying to find arbitrage but there are intances of this issues related to this PR whenever a sudden change or disconnection of network happened.

**Steps To Reproduce**
1. Setup Hummingbot client with gateway on AWS instance (since it would need a while to trigger)
2. Setup `perp` wallet 
3. Setup spot-perpetual-arbitrage (paper_trade for CEX would do) and use `perp` as derivative connector
4. Start the strategy and leave for a while or overnight
5. Check the client and docker logs 

**Release version**
Development branch dev-1.8.0

**Attachments**
[09122022.zip](https://github.com/hummingbot/hummingbot/files/9544085/09122022.zip)

**Comment by rapcmia** [comment]
Severity: P2
Bounty: 50000 (https://hummingbot.org/maintenance/bugs/)
Rationale: Bug related to a Perpetual protocol connector on gateway

**Comment by JeremyKono** [comment]
Assigning this PR to @vic-en. Thank you for your contribution!

**Comment by JeremyKono** [comment]
Hi @vic-en, can you please provide us an update of the progress of this bounty?

**Comment by vic-en** [comment]
@JeremyKono I'm still on this.
I encountered more errors than described.
Overall, I should be able to make a pr on or before the end of this week.

**Comment by vic-en** [comment]
https://github.com/hummingbot/hummingbot/pull/5885

**Comment by rapcmia** [comment]
Fix has been deployed to version 1.12.0
